test/lit/passes/outlining.wast

```diff
@@ -238,7 +238,62 @@
   )
 )
 
-;; Tests that outlining works correctly with If control flow
+;; Tests that outlining works correctly with if-condition
+(module
+  ;; CHECK:      (type $0 (func))
+
+  ;; CHECK:      (type $1 (func (result i32)))
+
+  ;; CHECK:      (global $global$1 (mut i32) (i32.const 100))
+  (global $global$1 (mut i32) (i32.const 100))
+  ;; CHECK:      (func $outline$ (type $1) (result i32)
+  ;; CHECK-NEXT:  (i32.eqz
+  ;; CHECK-NEXT:   (global.get $global$1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+
+  ;; CHECK:      (func $a (type $0)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (call $outline$)
+  ;; CHECK-NEXT:   (global.set $global$1
+  ;; CHECK-NEXT:    (i32.const 15)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $a
+    (if
+      (i32.eqz
+        (global.get $global$1)
+      )
+      (global.set $global$1
+        (i32.const 15)
+      )
+    )
+  )
+  ;; CHECK:      (func $b (type $0)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (call $outline$)
+  ;; CHECK-NEXT:   (global.set $global$1
+  ;; CHECK-NEXT:    (i32.const 20)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $b
+    (if
+      (i32.eqz
+        (global.get $global$1)
+      )
+      (global.set $global$1
+        (i32.const 20)
+      )
+    )
+  )
+)
+
+;; Outline if-true.
+;; TODO: Ideally outlining would keep the if-true inline in $outline$, instead
+;; of moving this to another outlined function ($outline$_3) because of the
+;; unique symbol between the if-condition and if-true.
```

Comment:
Looks like this should be moved to the "Outline if control flow, with matching if-condition, if-true, if-false" test below

Reply:
thanks!